app/lib/actions/action_with_sub_plans.rb

```diff
@@ -0,0 +1,156 @@
+module Actions
+
+  class Actions::ActionWithSubPlans < Actions::EntryAction
+
+    middleware.use Actions::Middleware::KeepCurrentUser
+    middleware.use Actions::Middleware::SetCurrentTaskId
+
+    SubPlanFinished = Algebrick.type do
+      fields! :execution_plan_id => String,
+              :success           => type { variants TrueClass, FalseClass }
```

Comment:
https://github.com/pitr-ch/algebrick/blob/master/lib/algebrick/types.rb#L31-L33 can be used instead